### DISCUSSION TITLE
Sunbird RC Learning Path

### DISCUSSION THREAD

**Discussion by coolbung** [Sunbird-RC/community#676] [General]
Proposing a learning path for anyone who wishes to get started with using or developing for Sunbird RC.

The learner will set up the below use case as part of the learning journey
1. Platform for a RTO to manage citizen and vehicle registry
2. Allow citizens to register
3. RTO officials add vehicle and associate to a citizen
4. When a vehicle needs to be transferred, RTO officials need to be able to transfer the vehicle to another citizen
5. When a vehicle is associated to a citizen, the citizen should receive a verifiable credential for the registration

*Core Sunbird Learning Path*
1. Start by defining the database design / schema for the above use case (Imagine you were building this in any RDBMS)
2. Create the schema json files required by Sunbird RC [schema example](https://rc.sunbird.org/use/developers-guide/schema-setup/schema-configuration)
3. Get to know the role based access implementation so that vehicle records can only be managed by RTO officials, and citizen records can be managed by citizens. For  this, the schemas need to have the appropriate [role configuration](https://rc.sunbird.org/use/developers-guide/schema-setup/schema-configuration#roles).
4. Create the verifiable credential related configuration [VC schema example](https://rc.sunbird.org/use/developers-guide/schema-setup/schema-configuration#credential-template)
5. Install sunbird RC using the registry-cli [installation instructions](https://rc.sunbird.org/use/getting-started/installation-guide)
6. Initialise the schemas 
7. Create a few users in Keycloak who will have the `rto-officials` role. 
8. View the auto generated swagger API docs [API](https://rc.sunbird.org/api-reference/other-apis/swagger-json-api)
9. Use the API to sign up as a citizen [API](https://rc.sunbird.org/api-reference/registry/create-an-entity)
10. Use the API to create a vehicle and associate to a citizen  [API](https://rc.sunbird.org/api-reference/registry/create-an-entity)
11. Use the API to update the citizen associated to a vehicle [API](https://rc.sunbird.org/api-reference/registry/update-an-entity)
12. View the VC json [API](https://rc.sunbird.org/api-reference/registry/get-an-entity-by-id)
13. Use the search APIs to search vehicles by citizen, or lookup a vehicle by it's registration number [API](https://rc.sunbird.org/api-reference/discovery-api/search-an-entity)

*Frontend Learning Path*
1. Configure the citizen sign up screen, and create a few citizen accounts
2. Configure the form for adding / updating a vehicle and add a few vehicles
3. Configure the search page and search for vehicles
4. Configure the details pages for citizen and vehicle

*Backend Learning Path*
https://www.youtube.com/watch?v=C-cmtsEuPU0

Do review and add any areas @gamemaker1 @chakshugautam @pallakartheekreddy @shiva-rakshith @snehal0904 @chaitrali-r @Renuka-S @suvarnakale @surendrasinghs 

@tejash-jl can you help with the backend learning path

Also, we'd need a similar outline for a development / customisation learning path. Request you to help with that too.

**Comment by coolbung** [comment]
@Renuka-S @Pratikshakhandagale -- please help with the links to the respective pages from the documentation

**Reply by ddevadat** [reply]
@coolbung need some help with the VC part of this learning part. Is there an learning video which explains how to create a custom credential template.

**Comment by srprasanna** [comment]
Here is some sample code to help you started with the [backend learning path](https://github.com/Sunbird-RC/bootcamp).

**Comment by ankit-apk** [comment]
I'm getting unauthorised error even when I'm generating access token for the entity and using the same for updating the entity data
